include/circt/Dialect/HW/HWOpInterfaces.td

```diff
@@ -32,79 +32,67 @@ def HWModuleLike : OpInterface<"HWModuleLike", [
   let description = "Provide common module information.";
 
   let methods = [
-    InterfaceMethod<"Get the number of ports",
-    "size_t", "getNumPorts">,
 
     InterfaceMethod<"Get a port symbol attribute",
     "::circt::hw::InnerSymAttr", "getPortSymbolAttr", (ins "size_t":$portIndex)>,
 
     InterfaceMethod<"Get the module type",
-    "::circt::hw::ModuleType", "getHWModuleType", (ins),
-    /*methodBody=*/[{}],
-    /*defaultImplementation=*/[{ return ::circt::hw::detail::fnToMod($_op, getInputNames(), getOutputNames()); }]>,
-
-    InterfaceMethod<"Return the names of the inputs to this module",
-    "mlir::ArrayAttr", "getInputNames", (ins),
-    /*methodBody=*/[{}],
-    /*defaultImplementation=*/[{
-      return $_op->template getAttrOfType<ArrayAttr>("argNames");
-    }]>,
+    "::circt::hw::ModuleType", "getHWModuleType", (ins)>,
+  ];
 
-    InterfaceMethod<"Return the names of the outputs this module",
-    "mlir::ArrayAttr", "getOutputNames", (ins),
-    /*methodBody=*/[{}],
-    /*defaultImplementation=*/[{
-      return $_op->template getAttrOfType<ArrayAttr>("resultNames");
-    }]>,
+  let extraSharedClassDeclaration = [{
 
-    InterfaceMethod<"Return the locations of the inputs to this module",
-    "mlir::ArrayAttr", "getInputLocs", (ins),
-    /*methodBody=*/[{}],
-    /*defaultImplementation=*/[{
-      return $_op->template getAttrOfType<ArrayAttr>("argLocs");
-    }]>,
+    size_t getNumPorts() {
```

Comment:
If these will live here for a while, maybe grab the dropped description for the interface methods and add as doxygen comments?

If they're documented on an interface that might be sufficient.